### ISSUE TITLE
Support for self-signed CA certificate

### ISSUE THREAD

**Issue by Ziris85** [simeonradivoev/PlayniteMQTTClient#7]
Hello!

I attempted to shift my MQTT connection in the playnite extension over to a TLS connection, where I have it configured with an SSL signed with a custom CA. I have this CA imported into my Windows certificate trust root, and browsers and other things will accept connections to local resources with certificates signed with this CA just fine. However, this addon will complain:

> MQTT: The remote certificate is invalid according to the validation procedure.

Might it be possible to adjust the addon to allow for connections to MQTT servers using self-signed SSL certificates?

Thanks for the addon!

**Comment by simeonradivoev** [comment]
yea should be fairly easy to point to a certificate file. Though I don't personally use them so can't easily test it.

**Comment by simeonradivoev** [comment]
I added the ability to specify the certificate path hopefully that was enough.

**Comment by Ziris85** [comment]
@simeonradivoev , thanks for the fast response! I updated and gave it a test, pointing it to my custom CA, but I'm still getting the same error message. I have done zero coding in C#, but I did find [this example snippet](https://github.com/dotnet/MQTTnet/blob/master/Samples/Client/Client_Connection_Samples.cs#L440) in the dotnet repo. Using that reference, might this make sense?
```
            if (settings.Settings.UseSecureConnection)
            {
                optionsUnBuilt = optionsUnBuilt.WithTlsOptions(o =>
                {
                    o.UseTls();
                    if (!string.IsNullOrEmpty(settings.Settings.CertificatePath) && File.Exists(settings.Settings.CertificatePath))
                    {
                      o.WithTlsOptions(new MqttClientTlsOptionsBuilder()
                        o.WithTrustChain(File.ReadAllBytes(settings.Settings.CertificatePath)) 
                          .Build());
                    }
                });
            }
```
Again, having done no coding with C# I don't know if this makes any sense or not heh, either in the broader context of your plugins code or C# in general...I'm happy to do some testing though!

**Comment by simeonradivoev** [comment]
You did check the `Secure TCP Connection` option on right?
Also do the certs have a password?

**Comment by simeonradivoev** [comment]
[MQTTClient_90c44048-4f8f-43f7-a0c1-f8164bf1d7ef_1_3_1.zip](https://github.com/simeonradivoev/PlayniteMQTTClient/files/14807968/MQTTClient_90c44048-4f8f-43f7-a0c1-f8164bf1d7ef_1_3_1.zip)
Try this change.

**Comment by Ziris85** [comment]
@simeonradivoev , yep, `Secure TCP Connection` is checked, and no passwords on certs/keys. I gave your zip a try but sadly I'm still getting the same message. Here's what my config looks like:

![image](https://github.com/simeonradivoev/PlayniteMQTTClient/assets/14392979/76a8494b-43bf-4b8b-858d-5e392fd8dc65)

I have a number of other devices connecting to this with the same connection parameters successfully, so the server itself SHOULD be OK. I did go nosing around the server logs just in case, and found nothing (which makes sense, since it doesn't look like the client is even negotiating the connection far enough to register on the server). I looked around for some weird quirks for C# or something that might help explain the issue, but didn't find anything. I did find [this](https://stackoverflow.com/questions/54938026/curl-unknown-error-0x80092012-the-revocation-function-was-unable-to-check-r) with a few suggestions to check antivirus. The suggestions were specific to AVG, and though I'm just using Defender I disabled that as well just for fun, but it didn't change anything. I AM seeing the same general behavior, though - browsers work fine, but curl throws that revocation error. idk if that's related at all though.

I've also confirmed (multiple times, lol) that the CA cert that I'm pointing to IS the right one. I went so far as to copy that "my-ca.crt" file onto a linux machine, pulled the certificate from the `rmq.lan:8883` server with `openssl s_client` and compared them:
```
~]$ openssl verify -verbose -CAfile my-ca.crt server.crt 
server.crt: OK
```

**Comment by Ziris85** [comment]
I downloaded [MQTT Explorer](https://mqtt-explorer.com/) and tested that way from the Windows machine, and the connection is also successful there if I provide that same ca file and "Validate certificate".
